plugins/postcss-focus-within/src/browser.js

```diff
@@ -0,0 +1,75 @@
+/* global document */
+import isValidReplacement from './is-valid-replacement.mjs';
+function generateHandler(replaceWith) {
+	let selector;
+	let remove;
+	let add;
+	const lastElements = [];
+
+	if (replaceWith[0] === '.') {
+		selector = replaceWith.slice(1);
+		remove = (el) => el.classList.remove(selector);
+		add = (el) => el.classList.add(selector);
+	} else {
+		// A bit naive
+		selector = replaceWith.slice(1, -1);
+		remove = (el) => el.removeAttribute(selector, '');
+		add = (el) => el.setAttribute(selector, '');
+	}
+
+	return function handleFocusChange() {
+		lastElements.forEach(lastElement => remove(lastElement));
+		lastElements.length = 0;
+
+		let activeElement = document.activeElement;
+
+		// only add focus if it has not been added and is not the document element
+		if (!/^(#document|HTML|BODY)$/.test(Object(activeElement).nodeName)) {
+			while (activeElement && activeElement.nodeType === 1) {
+				add(activeElement);
+				lastElements.push(activeElement);
+
+				activeElement = activeElement.parentNode;
+			}
+		}
+	};
+}
+
+export default function focusWithin(opts) {
+	// configuration
+	const options = Object.assign(
```

Comment:
`Object.assign` doesn't have good legacy support.
I will do another pass to check for API usage.

Reply:
Thanks for flagging this. You're completely right! Thanks for putting me in the right path :)